database/migrations/2025_09_05_000000_remove_matchmaking_pool_beatmaps_pk.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('matchmaking_pool_beatmaps', function (Blueprint $table) {
+            $table->dropPrimary('pool_id');
+            $table->index('pool_id');
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
```

Comment:
I was about to let this pass but noticed if I do a rollback (like on dev) and then rerun the migration, it'll fail due to missing primary key to be dropped and duplicated index to be created

Reply:
I'm not sure how to do this one... The simple:

```php
        Schema::table('matchmaking_pool_beatmaps', function (Blueprint $table) {
            $table->dropIndex('pool_id');
            $table->primary('pool_id');
        });
```

Results in:

```
SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP 'pool_id'; check that column/key exists (Connection: mysql, SQL: alter table `matchmaking_pool_beatmaps` drop index `pool_id`)
```

Plus, adding the primary would result in a potential PK violation when rolling back with items in the table, so I thought just not doing anything would be better...

Would appreciate guidance.

Reply:
Hmm, it works if the index is `['pool_id']` (array) instead. Is that acceptable?